src/Homomorphisms/Setoid/Basic.lagda

```diff
@@ -184,6 +305,40 @@ ker[_⇒_]_↾_ : (𝑨 : SetoidAlgebra α ρᵃ) (𝑩 : SetoidAlgebra β ρᵇ
  →            hom 𝑨 𝑩 → swelldef 𝓥 β → SetoidAlgebra _ _
 ker[ 𝑨 ⇒ 𝑩 ] h ↾ wd = kerquo 𝑨 𝑩 wd h
 
+
+
+-- setoid version
+
+module _ (𝑨 : SetoidAlgebra α ρᵃ) (𝑩 : SetoidAlgebra β ρᵇ) where
+ open SetoidAlgebra
+ open Setoid
+ open Func
+ private
+  A = 𝕌[ 𝑨 ]
+  B = 𝕌[ 𝑩 ]
+  ≈B = _≈_ (Domain 𝑩)
+ HomKerComp : (h : Hom 𝑨 𝑩) → 𝑨 ∣≈ (kerRel ≈B ∣ h ∣)
+ HomKerComp h f {u}{v} kuv = Goal
```

Comment:
Personally, I find the "equational reasoning" version of this proof much more readable than this style...

Reply:
I completely agree, but I've never used equational reasoning with Setoid equality...  I suspect it's just as easy as ≡-Reasoning, once I get the definitions imported.  Would you happen to have an example of that you could show me?

Reply:
The HomReasoning sub-module here https://agda.github.io/agda-categories/Categories.Category.Core.html shows how to set up some nice combinators. And then the rest of the library is full of examples of use.

Reply:
Great, thanks!  I'll check it out now.